src/plugins/BuildServerIntegration/GitlabIntegration/Settings/GitlabRemoteParser.cs

```diff
@@ -9,7 +9,7 @@ public partial class GitlabRemoteParser : RemoteParser
         [GeneratedRegex(@"git(?:@|://)(?<host>[^/]+)[:/](?<owner>[^/]+)/(?<repo>[\w_\.\-]+)\.git")]
         private static partial Regex GitlabSshUrlRegex();
 
-        [GeneratedRegex(@"https?://(?<host>[^@]+)/(?<owner>[^/]+)/(?<repo>[\w_\.\-]+)(?:.git)?")]
+        [GeneratedRegex(@"https?://(?<host>[^/@]+)/(?<owner>[^/].+)/(?<repo>[\w_\.\-]+)(?:.git)?")]
```

Comment:
Were the added '.' really intended?
```suggestion
        [GeneratedRegex(@"https?://(?<host>[^/@]+)/(?<owner>[^/]+)/(?<repo>[\w_\.\-]+)(?:.git)?")]
```

Reply:
after testing.
If we omit the dot (.) from the pattern, it inadvertently leads to incorrect namespace matching for repositories such as git.test.ai/aiot/cn010/reaction.git. The intended and correct namespace that should be captured is aiot/cn010,not aiot

Reply:
@redcatH Maybe it is a good idea to add your case in the unit tests `GitlabRemoteParserTests.cs`.

Reply:
So you say that owner can be multiple directory levels?
In that case you want a greedy regex that may include /.
Remove `[^/]` It makes no sense to just check first char.